opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -258,9 +261,9 @@ def _check_attribute_value_sequence(sequence: Sequence) -> Optional[str]:
         if first_element_type not in (bool, str, Number):
             return "invalid type"
 
-        for element in sequence:
-            if not isinstance(element, first_element_type):
-                return "different type"
+        if any(not isinstance(element, first_element_type) for element in sequence):
```

Comment:
How is this better than the previous implementation?

Reply:
Just a minor detail here. I think the original implementation may be _slightly_ better because it stops at the very first element in `sequence` that is not an instance of `first_element_type`. In the proposed implementation here, the _whole_ `sequence` is iterated through (I think it is, not completely sure), producing a list that then is passed to `any`.

Reply:
Makes sense. I have removed this implementation and left it as it was.